routes/api/v1/favorites.js

```diff
@@ -42,37 +43,93 @@ router.post('/', async function(req, res, next) {
   var favorite = await UserCity.findOrCreate({
     where: {
       UserId: user.id,
-      CityId: city.id
+      CityId: city[0].id
     },
     defaults: {
       UserId: user.id,
-      CityId: city.id
+      CityId: city[0].id
     }
-  })
+  });
 
   res.setHeader("Content-Type", "application/json");
-  if (user && user.apiKey === req.body.apiKey) {
+  if (user && favorite && user.apiKey === req.body.apiKey) {
     res.status(201).send(JSON.stringify({
-      message: `${city.name}, ${city.state}, ${city.country} has been added to your favorites.`
+      message: `${city[0].name}, ${city[0].state}, ${city[0].country} has been added to your favorites.`
     }));
   } else {
-    res.status(401).send(JSON.stringify({error: 'Something went wrong'}))
-  }
+    res.status(401).send(JSON.stringify({error: 'Something went wrong.'}))
+  };
 });
 
-router.get('/', function(req, res, next) {
-  User.findOne({
+router.get('/', async function(req, res, next) {
+  var user = await User.findOne({
     where: {
       apiKey: req.body.apiKey
     }
-  }).then(user => {
-    return user.getCities()
-  }).then(cities => {
-    res.setHeader("Content-Type", "application/json")
-    res.status(200).send(JSON.stringify({ favorites: cities }))
-  }).catch(error => {
+  });
+
+  const forecast = async (lat, long) => {
+    try {
+      const response = await fetch(`https://api.darksky.net/forecast/${process.env.DARKSKY_KEY}/${lat},${long}`);
+      const data = await response.json()
+      return data;
+    } catch (error) {
+      return error;
+    }
+  };
+
+  var cities = await user.getCities()
+  var favorites = []
+  for (let count = 0; count < cities.length; count++) {
```

Comment:
I struggled a little bit with this part myself. I found this article that might be helpful if you want to consider using map:
http://cryto.net/~joepie91/blog/2015/05/04/functional-programming-in-javascript-map-filter-reduce/
Takes a little less work to avoid the looping

Reply:
Thanks! I'll see if I can refactor to use `.map`